File: 백준/Silver/10974. 모든 순열/모든 순열.js
const fs = require('fs')
var arr, n;
var arr2 = Array()

function cal(cnt) {
  if (cnt == n) {
    console.log(arr2.join(' '));
    return
  }

  for (let i = 1; i <= n; i++) {
    if (!arr[i]) {
      arr[i] = true;
      arr2.push(i);
      cal(cnt + 1)
      arr[i] = false;
      arr2.pop();
    }
  }

}

fs.readFile('/dev/stdin', 'utf-8', (err, data) => {
  if (err) {
    console.error('Error reading the file : ', err);
    return;
  }
  n = Number(data)
  arr = Array(n + 1).fill(false);
  cal(0);
})

File: 백준/Silver/10974. 모든 순열/모든 순열.py
import sys

def cal() :
  if len(arr) == n :
    for i in arr : print(i,end=" ")
    print()
    return
  
  for i in range(n):
    if not check[i] :
      check[i] = True
      arr.append(data[i])
      cal()
      check[i] = False
      arr.pop()
      

if __name__ == "__main__":
  #sys.stdin = open('input.txt','r')
  n = int(sys.stdin.readline())
  data = [i+1 for i in range(n)]
  
  check = [False for i in range(n)]
  arr = []
  
  cal()